source/perflib/basename.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_PERFLIB_BASENAME_H
+# define OSSL_PERFLIB_BASENAME_H
+# pragma once
+
+extern const char *basename(const char *);
```

Comment:
I know this different to what's found everywhere. The reason I've decided to diverge is the return value in case path is NULL (or empty string). In this case the function does `return ".";` Therefore I prefer `const char *`.

Reply:
Then remove this special behavior? Do we need it for anything?

Reply:
another option would be to use static buffer and return it as a result. this is the closest to libc implementations. It does not need to be threadsafe. perftools call it from main thread only.

Reply:
IMO returning NULL or the empty string that was passed in is fine.

Reply:
I concur with @t8m: since this implementation is for Windows only (as I understand it), any callers must be able to cope with the behaviour of the POSIX functions (since that's the one they may get), so it doesn't help to give this one different behaviour, however useful it may be.